src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift

```diff
@@ -409,13 +417,90 @@ extension ExposureSubmissionCoordinator {
 		navigationController?.present(alert, animated: true, completion: nil)
 	}
 
+	
 	func showThankYouScreen() {
-		let vc = createSuccessViewController()
-		push(vc)
-	}
+		let thankYouVC = ExposureSubmissionThankYouViewController { [weak self] in
+			self?.showSymptomsScreen()
+		} onSecondaryButtonTap: { [weak self] in
+			self?.presentThankYouCancelAlert()
+		} presentCancelAlert: { [weak self] in
+			self?.presentThankYouCancelAlert()
+		}
 
-	// MARK: - Private
+		push(thankYouVC)
 
+	}
+	
+	// MARK: - Internal
+	
+	func presentPositiveTestResultCancelAlert() {
+		guard let navigationController = navigationController else {
+			Log.error("Can't present SubmissionSymptomsCancelAlert - missing navigationController")
+			return
+		}
+		
+		let alertTitle = isSubmissionConsentGiven ? AppStrings.ExposureSubmissionSymptomsCancelAlert.title : AppStrings.ExposureSubmissionPositiveTestResult.noConsentAlertTitle
+		let alertMessage = isSubmissionConsentGiven ? AppStrings.ExposureSubmissionSymptomsCancelAlert.message : AppStrings.ExposureSubmissionPositiveTestResult.noConsentAlertDescription
+		
+		let alertButtonLeft = isSubmissionConsentGiven ? AppStrings.ExposureSubmissionSymptomsCancelAlert.cancelButton :
+			AppStrings.ExposureSubmissionPositiveTestResult.noConsentAlertButtonLeft
+		
+		let alertButtonRight = isSubmissionConsentGiven ? AppStrings.ExposureSubmissionSymptomsCancelAlert.continueButton :
+			AppStrings.ExposureSubmissionPositiveTestResult.noConsentAlertButtonRight
+		
+		let alert = UIAlertController(
+			title: alertTitle,
+			message: alertMessage,
+			preferredStyle: .alert)
+		
+		alert.addAction(UIAlertAction(
+							title: alertButtonLeft,
+							style: .cancel,
+							handler: { [weak self] _ in
+								self?.dismiss()
+							})
+		)
+		
+		alert.addAction(UIAlertAction(
+							title: alertButtonRight,
+							style: .default)
+		)
+		navigationController.present(alert, animated: true, completion: nil)
+	}
+	
+	
+	func presentThankYouCancelAlert() {
```

Comment:
```suggestion
	private func presentThankYouCancelAlert() {
```

Reply:
It seems, that the whole coordinator has some inconsistencies with visibility states. Let me check the whole thing.

Reply:
Refactored to private 👍